trace/HTTP.md

```diff
@@ -90,18 +103,20 @@ Implementations SHOULD create message event when response size is determined.
 ## Attributes
 
 Implementations SHOULD set the following attributes on the client and server spans. For a server,
- request represents the incoming request. For a client, request represents the outgoing request.
+request represents the incoming request. For a client, request represents the outgoing request.
 
-All attributes are optional.
+All attributes are optional, but collector should make the best effort
+to collect those.
```

Comment:
can we ask to provide wither url or host and path? adding all of them seems redundant

Reply:
it is redundant. But it may help to set up metrics aggregation and build exporters easier. `Path` should be calculated anyway to construct `name`. `Host` is often used for aggregations and it will be extracted by exporter in most cases. So I'd suggest to keep all of them together. What is your biggest concern here - wire size or performance?

Reply:
I believe that Url parsing is relatively cheap and we are not going to gain much by duplication and avoiding parsing.

 My biggest concern is ambiguity and inconsistency.   I think our goal is to have all instrumentations to report the same data (when it's feasible). If we provide two different ways to report the same information, there will be some inconsistency as instrumentations would use whatever is more convenient, especially when there is obvious redundancy in the data.
Cost of changing the instrumentation in the backward compatible way is really high and is not justifiable, so we'll live with slightly different instrumentations and all kinds of workarounds on backends forever.

So, I'd prefer we have just one and straightforward way to set all http related attributes, but it would be great to hear what @adriancole and Jaeger and stack driver people think about it as they seem to rely on current attribute spec.

Reply:
I think what is collected and what is exported is the key issue. for example as a programming structure it matters less as you can derive host from url etc. some libraries you cannot get url (as you dont know scheme) so it is fake which might be ok.

problem really is when we simultaneously use same thing to talk about structs and then what is exported. I would prefer a policy where something claims the tags it needs. communicating this might be tricky now that collection tier is in scope of the project. for example who owns that config? does collector push data policy to the library?

Reply:
I understand you are advocating to collect all proposed. Exporter may decide to cut some?

It is very useful to have a min bar for which properties one should expect be collected for http.